pkg/cmd/workflow.go

```diff
@@ -0,0 +1,364 @@
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/puppetlabs/relay/pkg/errors"
+	"github.com/puppetlabs/relay/pkg/format"
+	"github.com/puppetlabs/relay/pkg/model"
+	"github.com/puppetlabs/relay/pkg/util"
+	"github.com/spf13/cobra"
+)
+
+func newWorkflowCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "workflow",
+		Short: "Manage your relay workflows",
+		Args:  cobra.MinimumNArgs(1),
+	}
+
+	cmd.AddCommand(newAddWorkflowCommand())
+	cmd.AddCommand(newReplaceWorkflowCommand())
+	cmd.AddCommand(newDeleteWorkflowCommand())
+	cmd.AddCommand(newRunWorkflowCommand())
+	cmd.AddCommand(newListWorkflowsCommand())
+	cmd.AddCommand(newDownloadWorkflowCommand())
+
+	return cmd
+}
+
+func doAddWorkflow(cmd *cobra.Command, args []string) error {
+	file, ferr := readFile(cmd)
+
+	if ferr != nil {
+		return ferr
+	}
+
+	workflowName, nerr := getWorkflowName(args)
+
+	if nerr != nil {
+		return nerr
+	}
+
+	Dialog.Progress("Creating your workflow...")
+
+	workflow, cwerr := Client.CreateWorkflow(workflowName)
+
+	if cwerr != nil {
+		return cwerr
+	}
+
+	revision, rerr := Client.CreateRevision(workflow.Workflow.Name, file)
+
+	if rerr != nil {
+
+		// attempt to revert creation of workflow record
+		Client.DeleteWorkflow(workflow.Workflow.Name)
+
+		return rerr
+	}
+
+	wr := model.NewWorkflowRevision(workflow.Workflow, revision.Revision)
+
+	wr.Output(Config)
+
+	Dialog.Infof(`Successfully created workflow %v
+			
+View more information or update workflow settings at %v`,
+		workflow.Workflow.Name,
+		format.GuiLink(Config, "/workflow/%v", workflow.Workflow.Name),
+	)
+
+	return nil
+}
+
+func newAddWorkflowCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add [workflow name]",
+		Short: "Add a relay workflow from a local file",
+		Args:  cobra.MaximumNArgs(1),
+		RunE:  doAddWorkflow,
+	}
+
+	cmd.Flags().StringP("file", "f", "", "Path to relay workflow file.")
+
+	return cmd
+}
+
+func doReplaceWorkflow(cmd *cobra.Command, args []string) error {
+	file, err := readFile(cmd)
+
+	if err != nil {
+		return err
+	}
+
+	workflowName, err := getWorkflowName(args)
+
+	if err != nil {
+		return err
+	}
+
+	Dialog.Info("Replacing workflow " + workflowName)
+
+	workflow, werr := Client.GetWorkflow(workflowName)
+
+	if werr != nil {
+		return werr
+	}
+
+	revision, rerr := Client.CreateRevision(workflowName, file)
+
+	if rerr != nil {
+		return rerr
+	}
+
+	wr := model.NewWorkflowRevision(workflow.Workflow, revision.Revision)
+
+	wr.Output(Config)
+
+	Dialog.Infof(`Successfully updated workflow %v
+			
+Updated configuration is visible at %v`,
+		workflowName,
+		format.GuiLink(Config, "/workflow/%v", workflowName),
```

Comment:
same here - unless there's something in the url handlers that's supposed to be working and isn't - i needed `/workflows/` on both prod and staging.